pkg/component/networking/vpn/shoot/shoot.go

```diff
@@ -743,6 +749,23 @@ func (v *vpnShoot) getEnvVars(index *int) []corev1.EnvVar {
 		},
 	)
 
+	if !gardencorev1beta1.IsIPv6SingleStack(v.values.ReversedVPN.IPFamilies) {
+		envVariables = append(envVariables,
+			corev1.EnvVar{
+				Name:  "SHOOT_POD_NETWORK",
+				Value: v.values.ShootPodNetwork,
+			},
+			corev1.EnvVar{
+				Name:  "SHOOT_SERVICE_NETWORK",
+				Value: v.values.ShootServiceNetwork,
+			},
+			corev1.EnvVar{
+				Name:  "SHOOT_NODE_NETWORK",
+				Value: v.values.ShootNodeNetwork,
+			},
+		)
+	}
```

Comment:
The environment variable names should probably have more specific names as well.